### PR TITLE
Player implementation

### LINKED ISSUE #35: Player implementation

### PR DESCRIPTION
**Pull request by Guptacos**

### CODE REVIEW

File: spotifython/player.py [hunk 1]
```diff
@@ -1,20 +1,27 @@
...
-from spotifython import Spotifython as sp
-
-from response import Response
-from typing import Union
-
-class Player():
-    ''' Interact with a user's playback, such as pausing / playing the current
+""" User class
+
+This class represents a Player object, which is tied to a User
```
Comment (spenceryu) [inline spotifython/player.py]: "which is tied to a User object" or "which is tied to a Spotify User"? To make it clear that this is tied to the Spotifython user class.

File: spotifython/player.py [hunk 2]
```diff
@@ -25,351 +32,780 @@ class Player():
 
         position: always an integer that represents milliseconds.
```
Comment (spenceryu) [inline spotifython/player.py]: nit: no need to say "always an integer"? "an integer" works fine?

File: spotifython/player.py [hunk 3]
```diff
@@ -25,351 +32,780 @@ class Player():
...
+        self._session = session
+        self._user = user
 
 
     # Format should be 'Player for user <%s>' with user_id
-    def __str__(self) -> str:
-        return ''
+    def __str__(self):
+        return 'Player for %s' % str(self._user)
+
```
Comment (spenceryu) [inline spotifython/player.py:71]: extra \n's?

File: spotifython/player.py [hunk 4]
```diff
@@ -25,351 +32,780 @@ class Player():
...
+                     market=const.TOKEN_REGION,
+                     return_none=False):
+        """ Helper function for the getter methods.
+        Wraps calling the player endpoint and handling a missing key.
 
-        response.contents():
+        Args:
+            key: the key to get from the currently playing context
+            market: used in track relinking
+            return_none: if True: returns None when no device available
```
Comment (spenceryu) [inline spotifython/player.py]: I think inverting this variable to be `should_raise_error` would make more sense; I had to think a bit for the connection of the variable to its function to click.

File: spotifython/player.py [hunk 5]
```diff
@@ -25,351 +32,780 @@ class Player():
...
+                         if False: raises SpotifyError when no device available
+
+        Returns:
+            None if there is no active device and return_none is True
+            The result of player_data[key] otherwise
+
+        Raises:
+            SpotifyError: if spotify returns an error or the key isn't found
```
Comment (spenceryu) [inline spotifython/player.py]: capitalize Spotify

**Review by spenceryu — DISMISSED** [review]

File: spotifython/player.py (hunk 3 above)
Comment (Guptacos) [reply spotifython/player.py:71]: I like having the double new lines, makes the code less compact

**Pull request merged**